index/models.py

```diff
@@ -81,30 +57,97 @@ class Response(models.Model):
+	# vote_resolution to true, false otherwise
+	def set_resolution (self,bool_val):
+		self.vote_resolution = bool_val
+
+	# If there was a referendum voted on, set 
+	# vote_referendum to true, false otherwise
+	def set_referendum (self,bool_val):
+		self.vote_referendum = bool_val
+
+	# If some other action was taken/ referred to another group,
+	# set refer_to_other to true, false otherwise
+	def set_other (self,bool_val):
+		self.refer_to_other = bool_val
+
+	# Add information for the senate investigation
+	def add_info_senate(self,info):
+		self.investigation_info = info
+		return self.investigation_info
+	# Add information on the committee assigned to help
+	def add_info_committee (self,info):
+		self.committee_info = info
+		return self.committee_info
+	# Add information on the resolution that was voted on
+	def add_info_resolution (self,info):
+		self.resolution_info = info
+		return self.resolution_info
+	# Add information on the referendum that wsa voted on
+	def add_info_referendum (self,info):
+		self.referendum_info = info
+		return self.referendum_info
+	# Add information on the other group that the issue was
+	# referred to
+	def add_info_other (self, info):
+		self.refer_other_info = info
+		return self.refer_other_info
+
+class Petition(models.Model):
+	title = models.CharField(max_length=200)	# Title of petition
+	description = models.CharField(max_length=4000)	# The paragraph description
+	ID = models.IntegerField(999999, primary_key=True)
+	archived = models.BooleanField(default=False)
+	hidden = models.BooleanField(default=False)
+				
+	created_date = models.DateTimeField(db_index=True, default=timezone.now())	# Files the date created
+	expected_sig = models.IntegerField(300)		# The expected signature to move to the next step
+
+	author = models.ForeignKey(User, on_delete=models.CASCADE, related_name='petitions')	# The author of the petition
+	tags = models.ManyToManyField(Tag, related_name='petitions')	# The tags, probably a max of 3
+	signatures = models.ManyToManyField(Signature, related_name='petitions')	# The signature
+
+	senate_response = models.ForeignKey(Response, on_delete=models.CASCADE, related_name='petitions')		# If the senate has responded , their answer
 
+	# Returns title when asked for the item
+	def __unicode__(self):
+		return self.title		
+  
+   	# Sets the hidden variable as true if we don't want it to
+   	# Be displayed in the main site
+    def set_hidden(self, bool_val):
+    	self.hidden = bool_val
+
+    # If the petition isn't active, we can set the archived variable as true
+    def set_archived(self, bool_val):
+    	self.archived = bool_val
+
+    # Adds a description to the basic petitions model, with a max length
+    # of 4000 words
+    def add_description(self, descript):
+    	self.description = descript
+    	return True
+    # Returns true if we have enough signatures on the petition
+    def check_enough_sigs(self):
+    	if self.signatures.count() >= self.expected_sig:
```

Comment:
Pylint is giving me this error: `Instance of 'ManyToManyField' has no 'count' member`.

When I try to test it, it throws an error because the database isn't configured. I think for now, we can keep it and merge it in. We can revisit it in the database config PR.

Reply:
Yeah I agree, let's just revisit later